packages/turbo-repository/rust/src/lib.rs

```diff
@@ -206,11 +244,29 @@ impl Workspace {
             .collect();
 
         // Create a ChangeMapper with no ignore patterns
-        let global_deps_package_detector =
-            GlobalDepsPackageChangeMapper::new(&self.graph, std::iter::empty::<&str>()).unwrap();
-        let mapper = ChangeMapper::new(&self.graph, vec![], global_deps_package_detector);
-        let lockfile_change = changed_lockfile.map(|s| Some(s.into_bytes()));
-        let package_changes = match mapper.changed_packages(hash_set_of_paths, lockfile_change) {
+        let change_detector = comparison
+            .is_some()
+            .then(|| {
+                Either::Left(
+                    GlobalDepsPackageChangeMapper::new(&self.graph, std::iter::empty::<&str>())
```

Comment:
I'm realize this isn't correct. We need to treat any "root" that isn't the lockfile as an "all packages" change.

Currently any change to a file that doesn't belong to a package will result in only the "root" getting triggered: https://github.com/vercel/turborepo/blob/main/crates/turborepo-repository/src/change_mapper/package.rs#L130

We could change "root" handling for consumers, but that feels a little leaky